From e6d6c29e007b7710d83a3ec410425123457478de Mon Sep 17 00:00:00 2001
From: Prasanna S <prasannas@ami.com>
Date: Tue, 19 Aug 2025 09:27:09 +0530
Subject: [PATCH] Disable Active and BackupImage checkboxes for FVP

FVP firmware updates only the non-running image and switches it to the
running image during BMC reboot. It does not support updating both
images simultaneously or independently.

Upstream-Status: Inappropriate

Signed-off-by: Prasanna S <prasannas@ami.com>
Signed-off-by: Mohammed Javith Akthar M <mohammedjavitham@ami.com>
---
 src/store/modules/Operations/FirmwareStore.js | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/store/modules/Operations/FirmwareStore.js b/src/store/modules/Operations/FirmwareStore.js
index 802d7f79..1fef4bb7 100644
--- a/src/store/modules/Operations/FirmwareStore.js
+++ b/src/store/modules/Operations/FirmwareStore.js
@@ -148,7 +148,8 @@ const FirmwareStore = {
               data?.['@odata.id'] ===
               '/redfish/v1/UpdateService/FirmwareInventory/bmc_active'
             ) {
-              bmcActiveFeatureEnabled = true;
+              bmcActiveFeatureEnabled = false;
+              //Arm Fvp doesn't have active image support.so disable Active image checkbox
             }
             commit('setBmcActiveFeatureEnabled', bmcActiveFeatureEnabled);
             // Check if bmc_bkup is available
@@ -156,7 +157,8 @@ const FirmwareStore = {
               data?.['@odata.id'] ===
               '/redfish/v1/UpdateService/FirmwareInventory/bmc_bkup'
             ) {
-              bmcBackupEnabled = true;
+              bmcBackupEnabled = false;
+              //Arm Fvp doesn't have active image support.so disable Backup image checkbox
             }
             commit('setBmcBackupEnabled', bmcBackupEnabled);
             if (firmwareType === 'bmc') {
-- 
2.43.0

